curvefs/src/metaserver/partition_cleaner.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+ *  Copyright (c) 2021 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * @Project: curve
+ * @Date: 2021-12-15 10:54:37
+ * @Author: chenwei
+ */
+
+#include "curvefs/src/metaserver/partition_cleaner.h"
+#include <list>
+#include "curvefs/src/metaserver/copyset/meta_operator.h"
+
+namespace curvefs {
+namespace metaserver {
+
+bool PartitionCleaner::ScanPartition() {
+    if (partition_->GetInodeNum() == 0) {
+        LOG(INFO) << "Inode num is 0, delete partition from metastore";
+        MetaStatusCode ret = DeletePartition();
+        if (ret == MetaStatusCode::OK) {
+            return true;
+        } else {
+            LOG(WARNING) << "delete partition from copyset fail, ret = "
+                         << MetaStatusCode_Name(ret);
+            return false;
+        }
+    }
+
+    if (!copysetNode_->IsLeaderTerm()) {
+        return false;
+    }
+
+    std::list<uint64_t> InodeIdList;
+    partition_->GetInodeIdList(&InodeIdList);
+    for (auto inodeId : InodeIdList) {
```

Comment:
Need sleep before delete next inode. If thousands of inodes are deleted all the time, the pressure will be greater.

Reply:
done

Reply:
done